libcloudforensics/providers/aws/internal/ec2.py

```diff
@@ -566,3 +574,99 @@ def GenerateSSHKeyPair(self, vm_name: str) -> Tuple[str, str]:
               exception), __name__) from exception
     # If the call was successful, the response contains key information
     return key['KeyName'], key['KeyMaterial']
+
+  def CreateIsolationSecurityGroup(self,
+      vpc: str,
+      ingress_subnets: Optional[List[str]] = None
+    ) -> str:
+    """Creates an isolation security group.
+
+    Args:
+      vpc (str): VPC ID for the security group.
+      ingress_subnets (List[str]): Optional. Subnets to allow ingress from.
+
+    Returns:
+      str: The ID of the newly created security group.
+
+    Raises:
+      ipaddress.AddressValueError: If an invalid subnet is provided.
+      errors.ResourceCreationError: If an invalid VPC ID is provided.
+    """
+    client = self.aws_account.ClientApi(common.EC2_SERVICE)
+    description = 'Quarantine Security Group'
+    name = 'quarantine-{0:d}'.format(random.randint(10**(9),(10**10)-1))
+
+    # Test the subnets are well formed
+    if ingress_subnets:
+      for subnet in ingress_subnets:
+        ipaddress.IPv4Network(subnet, False)
+
+    # Create the SG
+    try:
+      sg_info = client.create_security_group(
+        Description=description,
+        GroupName=name,
+        VpcId=vpc)
+    except client.exceptions.ClientError as exception:
+      raise errors.ResourceCreationError(
+        'Could not create security group: {0!s}'.format(
+           exception), __name__) from exception
+
+    # SG's automatically include an everything outbound rule - remove it
+    sg_rules = client.describe_security_group_rules(
```

Comment:
Do any of these calls (`clients.xxx`) throw an exception? If so, wrap them.

Reply:
My understanding is it will only throw a permission denied (which isn't considered anywhere else either) or a ResourceNotFound - but we just created the resource.

Reply:
ACK, thanks for confirming